libV2/tasks/DiskUsage.js

```diff
@@ -16,29 +25,147 @@ class MonitorDiskUsage extends BaseTask {
+            return;
+        }
+
+        const oldestTimestamp = resp.result[0];
+        if (oldestTimestamp === -1) {
+            moduleLogger.error('No records found! nothing to delete!');
+            return;
+        }
+
+        const endTimestamp = oldestTimestamp + expirationBlockMicroSecs - 1;
+        moduleLogger.info(`deleting oldest ${config.diskUsage.expirationBlockSize}hr block of metrics`, {
+            start: oldestTimestamp, stop: endTimestamp,
+        });
+
+        await this._warp10.delete({
+            className: '~.*',
+            start: oldestTimestamp - 1,
+            end: endTimestamp,
+        });
+    }
+
+    async _execute(timestamp) {
         if (!this._enabled) {
             moduleLogger.debug('disk usage monitoring not enabled, skipping check');
             return;
         }
 
-        let size;
-        try {
-            size = await this._getUsage();
-        } catch (error) {
-            moduleLogger.error(`error calculating disk usage for ${this._path}`, { error });
-            return;
+        let size = null;
+        if (!this.isLeader) {
+            try {
+                size = await this._getUsage();
+            } catch (error) {
+                moduleLogger.error(`error calculating disk usage for ${this._path}`, { error });
+                return;
+            }
+            moduleLogger.debug(`using ${formatDiskSize(size)}`, { usage: size });
         }
 
-        moduleLogger.debug(`using ${formatDiskSize(size)}`, { usage: size });
+
+        if (this._softLimit !== undefined) {
+            let shouldDelete = false;
+            if (this._mode === 'local') {
+                moduleLogger.debug('Operating in local mode, only checking the current node');
+                shouldDelete = await this._checkSoftLimit(size, this.nodeId);
+            } else if (this._mode === 'distributed') {
+                if (this.isLeader) {
+                    try {
+                        const resp = await this._warp10.fetch({
+                            className: 'utapi.disk.monitor',
+                            start: 'now',
+                            stop: -1,
+                        });
+
+                        assert.notStrictEqual(resp.result, undefined);
+                        assert.notStrictEqual(resp.result.length, 0);
+
+                        if (resp.result[0] === '') {
+                            moduleLogger.warn('no disk usage entries found');
+                        } else {
+                            shouldDelete = JSON.parse(resp.result)
+                                .map(val => ({ node: val.l.node, used: val.v[0][1] }))
+                                .map(val => this._checkSoftLimit(val.used, val.node))
+                                .some(val => val);
+                        }
+                    } catch (error) {
+                        moduleLogger.error('error fetching disk usage data from warp 10', { error });
+                    }
```

Comment:
not sure how feasible it is but neither catch in this method has tests, or at least I am guessing so as I do not see tests with `assert.throws`.

Reply:
I can stub it with `sinon`